frontend/src/components/TaskItem/index.tsx

```diff
@@ -9,32 +10,69 @@ import Typography from '@mui/material/Typography';
 import Task from '../../types/Task';
 import ListItemText from '@mui/material/ListItemText';
 import LinearProgress, { LinearProgressProps } from '@mui/material/LinearProgress';
-import { Alert, Box, Button, ListItem, ListItemIcon, ListItemSecondaryAction, Snackbar } from '@mui/material';
+import { Alert, Box, ListItem, Snackbar } from '@mui/material';
 import Grid from '@mui/material/Unstable_Grid2';
 import TaskService from '../../services/TaskService';
 
 interface GetTaskFormProps {
   task: Task;
 }
 
-function LinearProgressWithLabel(props: LinearProgressProps & { value: number }) {
-  return (
-    <Box sx={{ display: 'flex', alignItems: 'center' }}>
-      <Box sx={{ width: '100%', mr: 1 }}>
-        <LinearProgress variant="determinate" {...props} />
-      </Box>
-      <Box sx={{ minWidth: 35 }}>
-        <Typography variant="body2" color="text.secondary">{`${Math.round(
-          props.value
-        )}%`}</Typography>
-      </Box>
-    </Box>
-  );
-}
-
 export default function TaskListItem(props: GetTaskFormProps) {
   const [open, setOpen] = React.useState(false);
 
+  function LinearProgressWithLabel(progressProps: LinearProgressProps & { value: number }) {
```

Comment:
why move this?

Reply:
so that `props.task` is visible to it

Reply:
i think its preferable to create a props interface for each subcomponent defined and then put the functions outside of the component function. otherwise the functions get re-created with the correct context on every render